### PR TITLE
Upgrade to OpenTracing-Java 0.30.0 and use the Active Span (in-proces…

### PR DESCRIPTION
**Pull request by objectiser**
…s propagation) mechanism

**DO NOT MERGE YET**

Currently uses snapshot version of the java-web-servlet-filter (need to build https://github.com/opentracing-contrib/java-web-servlet-filter/pull/25) and master for opentracing-java to include the Tags/BaseSpan fix.

Main change has been to remove the additional Span that gets created when there is no active span, and appears to be where the spring interceptor is called after the request has been handled and (for example) thrown an exception.

This additional Span was only being created for the boot tests (not mvc) so by not creating this additional Span is makes the tests consistent across spring boot and mvc. If this additional Span provides useful information in realworld scenarios, then we could create a separate PR to add it back in - but I think the current PR simplifies the mechanism and should cater for most cases.

Haven't re-reviewed properly, but wanted to provide the current status, as ideally would like to get this and the web-servlet-filter PRs ready to merge once OT-J 0.30.0 is released.

### CODE REVIEW

File: opentracing-spring-web-itest/mvc/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/mvc/MVCJettyITest.java [hunk 1]
```diff
@@ -68,129 +55,4 @@ protected TestRestTemplate getRestTemplate() {
         return testRestTemplate;
     }
 
-    @Test
-    public void testFilterException() throws Exception {
```
Comment (pavolloffay) [inline opentracing-spring-web-itest/mvc/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/mvc/MVCJettyITest.java]: I want these tests back we spent a lot of time handling corner cases to trace all possible error states.

Comment (objectiser) [inline opentracing-spring-web-itest/mvc/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/mvc/MVCJettyITest.java]: These tests are exact copies of the tests they override in the `AbstractBaseITests.java`, except that instead of handling two spans they were just handling one - the second being this additional `Span` that gets created when the interceptor is called outside the scope of the server span.

As mentioned in the description for this PR, I did not see the benefit of representing this span, and without it, the spring boot and mvc tests were exactly the same.

Can you explain why this extract Span is of value, when it occurs outside the scope of the server span? I can add the mechanism to capture this additional Span back in, but wanted to be sure it is worth it.

Comment (pavolloffay) [inline opentracing-spring-web-itest/mvc/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/mvc/MVCJettyITest.java]: This PR reduces level of visibility what is happening, it removes tracing of the standard error handling in spring boot.

**Comment by objectiser** [comment]
Closing as needs more thought.
